File: VB/Scripts/ErrorHandler.vb
Option Strict On
Option Explicit On

Imports System.Environment
Imports System.Windows.Forms

Namespace Scripts

    Public Class ErrorHandler

        Public Shared Sub DisplayMessage(ex As Exception)
            Dim sf As New System.Diagnostics.StackFrame(1)
            Dim caller As System.Reflection.MethodBase = sf.GetMethod()
            Dim currentProcedure As String = (caller.Name).Trim()
            Dim currentFileName As String = "" 'AssemblyInfo.GetCurrentFileName()
            Dim errorMessageDescription As String = ex.ToString()
            errorMessageDescription = System.Text.RegularExpressions.Regex.Replace(errorMessageDescription, "\r\n+", " ")
            Dim msg As String = "Contact your system administrator. A record has been created in the log file." + Environment.NewLine
            msg += (Convert.ToString("Procedure: ") & currentProcedure) + Environment.NewLine
            msg += "Description: " + ex.ToString() + Environment.NewLine
            MessageBox.Show(msg, "Unexpected Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])

        End Sub

        Public Shared Function RemoveString(ByVal text As String, Optional ByVal charValue As String = ".") As String
            Try
                If text.Contains(charValue) Then
                    Return text.Substring(0, text.IndexOf("."))
                Else
                    Return text
                End If

            Catch ex As Exception
                Return text

            Finally

            End Try

        End Function

        Public Shared Function ValidateString(ByVal text As String) As String
            Try
                If String.IsNullOrEmpty(text) Then
                    text = String.Empty
                Else
                    text = text.Replace(vbCr, "").Replace(vbLf, "")
                    text = text.Replace(",", ";")
                    text = text.Trim
                End If
                Return text

            Catch ex As Exception
                Return text

            Finally

            End Try

        End Function

    End Class

End Namespace

File: CS/Scripts/ErrorHandler.cs
using System;
using System.Windows.Forms;
using System.Runtime.InteropServices;
using log4net;
using log4net.Config;

[assembly: log4net.Config.XmlConfigurator(Watch = true)]

// <summary> 
// This namespaces if for generic application classes
// </summary>
namespace ShapeExtract.Scripts
{
    /// <summary> 
    /// Used to handle exceptions
    /// </summary>
    public class ErrorHandler
    {
        private static readonly ILog log = LogManager.GetLogger(typeof(ErrorHandler));

        /// <summary> 
        /// Used to produce an error message and create a log record
        /// <example>
        /// <code lang="C#">
        /// ErrorHandler.DisplayMessage(ex);
        /// </code>
        /// </example> 
        /// </summary>
        /// <param name="ex">Represents errors that occur during application execution.</param>
        /// <remarks></remarks>
        public static void DisplayMessage(Exception ex)
        {
            System.Diagnostics.StackFrame sf = new System.Diagnostics.StackFrame(1);
            System.Reflection.MethodBase caller = sf.GetMethod();
            string currentProcedure = (caller.Name).Trim();
            string errorMessageDescription = ex.ToString();
            errorMessageDescription = System.Text.RegularExpressions.Regex.Replace(errorMessageDescription, @"\r\n+", " "); //the carriage returns were messing up my log file
            string msg = "Contact your system administrator. A record has been created in the log file." + Environment.NewLine;
            msg += "Procedure: " + currentProcedure + Environment.NewLine;
            msg += "Description: " + ex.ToString() + Environment.NewLine;
            log.Error("[PROCEDURE]=|" + currentProcedure + "|[USER NAME]=|" + Environment.UserName + "|[MACHINE NAME]=|" + Environment.MachineName + "|[DESCRIPTION]=|" + errorMessageDescription);
            MessageBox.Show(msg, "Unexpected Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="text"></param>
        /// <param name="charValue"></param>
        /// <returns></returns>
        public static string RemoveString(string text, string charValue = ".")
        {
            try
            {
                if (text.Contains(charValue))
                {
                    return text.Substring(0, text.IndexOf("."));
                }
                else
                {
                    return text;
                }
            }
            catch (Exception)
            {
                return text;
            }
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="text"></param>
        /// <returns></returns>
        public static string ValidateString(string text)
        {
            try
            {
                if (string.IsNullOrEmpty(text))
                {
                    text = string.Empty;
                }
                else
                {
                    text = string.Empty;  //fix this
                    //text = text.Replace(vbCr, "").Replace(vbLf, "");
                    //text = text.Replace(",", ";");
                    //text = text.Trim;
                }

                return text;
            }
            catch (Exception)
            {
                return text;
            }

        }

    }
}